server/controllers/users-controller.js

```diff
@@ -0,0 +1,255 @@
+        ]
+      }).then(theUser =>
+        res.status(200).send({
+          done: true,
+          user: theUser.dataValues
+        })
+      );
+    } else {
+      UsersController.returnUnAuthroized(res);
+    }
+  }
+
+  /**
+   * method updateAUser to update user details
+   * @param {object} req - request details
+   * @param {object} res - response details
+   * @return {object} new userObject details;
+   */
+  static updateAUser(req, res) {
+    if (req.decoded.id === parseInt(req.params.id, 10)) {
+      db.user.update(req.body, {
+        where: {
+          id: req.decoded.id
+        },
+        fields: ['firstName', 'lastName', 'username', 'email',
+          'password', 'roleId'],
+        returning: true
+      }).then(updatedUser =>
+        res.status(200).send({
+          done: true,
+          user: UsersController.fetchUser(updatedUser[1][0].dataValues)
+        })
+      ).catch(error =>
+        res.status(400).send({
+          done: false,
+          message: error.errors[0].message
+        })
+      );
+    } else {
+      UsersController.returnUnAuthroized(res);
+    }
+  }
+
+  /**
+   * method returnUnAuthroized is to return Unauthorized message
+   * @param {object} res; the response obect
+   * @returns {object} response sent to the frontend
+   */
+  static returnUnAuthroized(res) {
+    return res.status(401).send({
+      done: false,
+      message: 'Unauthorized request'
+    });
+  }
+
+  /**
+   * method deleteUser to update user details
+   * @param {object} req - request details
+   * @param {object} res - response details
+   * @return {object} details of the deletion;
+   */
+  static deleteUser(req, res) {
+    if (req.decoded.id === parseInt(req.params.id, 10)) {
+      db.user.destroy({ where: { id: req.decoded.id } }).then(() =>
+        res.status(200).send({ done: true })
+      ).catch(error =>
+        res.status(400).send({
+          done: false,
+          message: error.errors[0].message
+        })
+      );
+    } else {
+      UsersController.returnUnAuthroized(res);
+    }
+  }
+
+  /**
+   * method getUserRole to get the role of a user
+   * @param {integer} userId - id of the user
+   * @param {function} callback function
+   * @return {function} callback function
+   */
+  static getUserRole(userId, callback) {
+    db.user.findAll({
+      where: {
+        id: userId
+      },
+      include: [db.role]
+    }).then(userInfo =>
+      callback(userInfo[0].dataValues.role.title)
+    ).catch(() =>
+      callback(false)
+    );
+  }
+
+  /**
+   * method getUserDocuments to get a user documents
+   * @param {object} req - request details
+   * @param {object} res - response details
+   * @return {array} array of documents;
+   */
+  static getUserDocuments(req, res) {
+    const userId = parseInt(req.params.id, 10);
+    db.document.findAll({ ownerId: userId, order: [['createdAt', 'DESC']] })
+      .then(documents => res.status(200).send({ doc: documents }));
```

Comment:
using `res.send` means the application will not be able to convert non JSON data, like null and undefined. You might want to consider using `res.json` instead